### DISCUSSION TITLE
Mutate Extension (WriteableSession) in Response

### DISCUSSION THREAD

**Discussion by michaelmerrill** [tokio-rs/axum#1534] [Q&A] [answered]
Is there a way I can access `axum_sessions::extractors::WritableSession` from `into_response`? The following is what I've tried, but it doesn't seem to update the session.

```(rust)
impl IntoResponse for LoginError {
    fn into_response(self) -> Response {
        let mut response = Redirect::to("/login").into_response();
        response
            .extensions_mut()
            .get_mut::<WritableSession>()
            .map(|session| session.insert("_login_error", self.to_string()));
        response
    }
}
```

**Comment by davidpdrsn** [comment, marked as answer]
The response you create with `Redirect::to("/login").into_response()` is a completely new response that doesn't know anything about the corresponding request, so it wont have any extensions.

Shoulds like you're looking for a middleware https://docs.rs/axum/latest/axum/middleware/fn.from_fn.html

**Reply by michaelmerrill** [reply]
Thank you!
